scripts/queuestat

```diff
@@ -177,42 +220,78 @@ class Queuestat(object):
 
             cnstat_fqn_file_name = cnstat_fqn_file + port
             if os.path.isfile(cnstat_fqn_file_name):
                 try:
                     cnstat_cached_dict = pickle.load(open(cnstat_fqn_file_name, 'rb'))
-                    print(port + " Last cached time was " + str(cnstat_cached_dict.get('time')))
-                    self.cnstat_diff_print(port, cnstat_dict, cnstat_cached_dict)
+                    if json_opt:
+                        json_output[port].update({"cached_time":cnstat_cached_dict.get('time')})
+                        json_output.update(self.cnstat_diff_print(port, cnstat_dict, cnstat_cached_dict, json_opt))
+                    else:
+                        print(port + " Last cached time was " + str(cnstat_cached_dict.get('time')))
+                        self.cnstat_diff_print(port, cnstat_dict, cnstat_cached_dict, json_opt)
                 except IOError as e:
                     print(e.errno, e)
             else:
-                self.cnstat_print(port, cnstat_dict)
+                if json_opt:
+                    json_output.update(self.cnstat_print(port, cnstat_dict, json_opt))
+                else:
+                    self.cnstat_print(port, cnstat_dict, json_opt)
 
-    def get_print_port_stat(self, port):
+        if json_opt:
+            print(json_dump(json_output))
+
+    def get_print_port_stat(self, port, json_opt):
+        """
+        Get stat for the port
+        If JSON option is True  print data in JSON format
+        """
         if not port in self.port_queues_map:
             print("Port doesn't exist!", port)
             sys.exit(1)
 
         # Get stat for the port queried
         cnstat_dict = self.get_cnstat(self.port_queues_map[port])
         cnstat_fqn_file_name = cnstat_fqn_file + port
+        json_output = {}
+        json_output[port] = {}
         if os.path.isfile(cnstat_fqn_file_name):
             try:
                 cnstat_cached_dict = pickle.load(open(cnstat_fqn_file_name, 'rb'))
-                print("Last cached time was " + str(cnstat_cached_dict.get('time')))
-                self.cnstat_diff_print(port, cnstat_dict, cnstat_cached_dict)
+                if json_opt:
+                    json_output[port].update({"cached_time":cnstat_cached_dict.get('time')})
+                    json_output.update(self.cnstat_diff_print(port, cnstat_dict, cnstat_cached_dict, json_opt))
+                else:
+                    print("Last cached time was " + str(cnstat_cached_dict.get('time')))
+                    self.cnstat_diff_print(port, cnstat_dict, cnstat_cached_dict, json_opt)
             except IOError as e:
                 print(e.errno, e)
         else:
-            self.cnstat_print(port, cnstat_dict)
+            if json_opt:
+                json_output.update(self.cnstat_print(port, cnstat_dict, json_opt))
+            else:
+                self.cnstat_print(port, cnstat_dict, json_opt)
```

Comment:
can we avoid this if..else above and here and do all processing within cnstat_print/cnstat_diff_print

Reply:
adding inside  cnstat_print/cnstat_diff_print will complicate logic because  cnstat_print/cnstat_diff_print print for each port, where as for JSON we should collect data for all ports and then print the json output. Checkout "get_print_all_stat".